openraft/src/replication/mod.rs

```diff
@@ -315,13 +315,15 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> Replication
     /// configured heartbeat interval.
     #[tracing::instrument(level = "debug", skip(self))]
     async fn send_append_entries(&mut self) -> Result<(), ReplicationError<C>> {
+        // first reset need_to_replicate flag
+        self.need_to_replicate = false;
```

Comment:
It should not clear the flag at first.

If there is an error return, without sending out any log entries, this flag should be left `true`.